src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -21,12 +21,12 @@ public static class AttachedProperties
             typeof(GeometryModel3D),
             new PropertyMetadata(false, ShowSelectedPropertyChanged));
 
-        public static void SetShowSelected(UIElement element, bool value)
+        public static void SetShowSelected(DependencyObject element, bool value)
```

Comment:
even though this references a less derived base type - this is still a binary break, and a missing method exception will be thrown for anything that called this previously and is not recompiled AFAIK, I don't know enough to say if it could even cause reference errors for users who reference DyanmoCoreWPF.

I think the most conservative pattern and the one we generally follow - is to avoid changes like this - obsolete the methods - and add a series of new ones - the implementations can be moved to the new methods, and the old methods can be wrappers.

If that is too much work, then extensive testing is required to see in (if) which situations this change causes runtime failures.

Reply:
These methods are only ever called with Helix objects and the base class of those Helix objects has changed. I think these changes have to be made in a breaking fashion/major release.